src/arch/ucx/machine.C

```diff
@@ -634,9 +641,11 @@ void LrtsExit(int exitcode)
 
     for (i = 0; i < ucxCtx.numRxReqs; ++i) {
         req = ucxCtx.rxReqs[i];
-        CmiFree(req->msgBuf);
-        ucp_request_cancel(ucxCtx.worker, req);
-        ucp_request_free(req);
+        if (!req->completed) {
```

Comment:
Why aren't we freeing `req->msgBuf` if the request is completed?

Reply:
it should be done in `handleOneRecvedMsg`, right?

Reply:
Ah, I see. Yes, it does happen at a higher level. So, this is just taking care of the incomplete ones. Got it.